pandas/tests/test_generic.py

```diff
@@ -1717,6 +1717,10 @@ def test_squeeze(self):
         p4d = tm.makePanel4D().reindex(labels=['label1'],items=['ItemA'])
         tm.assert_frame_equal(p4d.squeeze(),p4d.ix['label1','ItemA'])
 
+        # don't fail with 0 length dimensions
+        empty=pd.DataFrame(pd.np.empty(shape=(0,1)))
```

Comment:
test with all shapes here (series.frame,etc)

furthermore this should preserve name